client/src/utils/curriculum-helpers.js

```diff
@@ -0,0 +1,71 @@
+import { parse } from '@babel/parser';
+import generate from '@babel/generator';
+
+const removeHtmlComments = str => str.replace(/<!--(.|\s)*?-->/g, '');
+
+const removeCssComments = str => str.replace(/\/\*[\s\S]+?\*\//g, '');
+
+const removeJSComments = codeStr => {
+  // Note: removes trailing new lines and tailing spaces at end of lines
+  const options = {
+    comments: false,
+    retainLines: true,
+    compact: false,
+    concise: false,
+    minified: false
+  };
+  try {
+    const ast = parse(codeStr);
+    const { code } = generate(ast, options, codeStr);
+    return code;
+  } catch (err) {
+    return codeStr;
+  }
+};
+
+// TODO: Need to write tests before using this function
+const removeScriptElementComments = codeStr => {
+  try {
+    let sanitizedCode = codeStr.replace(
+      /<script[\s\S]*?>[\s\S]*?<\/script>/gi,
+      ''
+    );
+    codeStr.match(/<script[\s\S]*?>[\s\S]*?<\/script>/gi).forEach(script => {
+      const scriptTags = script.match(/(<([^>]+)>)/gi);
+      const jsCode = script.replace(/(<([^>]+)>)/gi, '');
+      sanitizedCode +=
+        scriptTags[0] + removeJSComments(jsCode) + '\n' + scriptTags[1];
+    });
+    return sanitizedCode;
```

Comment:
```suggestion
    let noScripts = codeStr.replace(
      /<script[\s\S]*?>[\s\S]*?<\/script>/gi,
      ''
    );
    codeStr.match(/<script[\s\S]*?>[\s\S]*?<\/script>/gi).forEach(script => {
      const scriptTags = script.match(/(<([^>]+)>)/gi);
      const jsCode = script.replace(/(<([^>]+)>)/gi, '');
      noScripts +=
        scriptTags[0] + removeJSComments(jsCode) + '\n' + scriptTags[1];
    });
    return noScripts;
```
We're not sanitizing here, so we shouldn't imply that we are.

Reply:
Shouldn't the variable name be something more like `noScriptElementComments` instead of `noScripts`?